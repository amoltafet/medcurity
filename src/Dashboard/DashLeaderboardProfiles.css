.cardBackgroundDash {
    margin-bottom: 2%;
    width: 100%;
    height: 22%;

}

.dashLeaderNumberPosition {
    color: #1e5b88;
    font-size: 90%;
    margin-top: 8%;
}

.profileImageDash {    
    height: 70%;
    margin-top: 5%;
}

.userTextDash {
    color: #1e5b88;
    font-size: 70%;
    font-family: Lato,sans-serif;
    text-overflow: ellipsis;
    margin-top: 10%;
}

.scoreTextDash {
    color: #127ebf;
    font-family: Lato,sans-serif;
    font-size: 70%;
    text-overflow: ellipsis;
    margin-top: 10%;
}

.pointsDash {
    color: #93c2ec;
    font-family: Lato,sans-serif;
    font-size: 70% ;
    text-overflow: ellipsis;
    margin-top: 10%;
}


.statusImageDash {
    width: 10%;
    height: 80%;
    margin-left: 6%;
    margin-bottom: 1%;
    margin-top: 1.5%;
}

