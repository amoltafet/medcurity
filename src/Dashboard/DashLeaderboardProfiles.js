// <div>Icons made by <a href="https://www.flaticon.com/authors/kliwir-art" title="kliwir art">kliwir art</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>
import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css'
import { Card, Image,  Col, Row } from 'react-bootstrap';
import axios from 'axios';
import { useEffect, useState } from "react";
import './DashLeaderboardProfiles.css'
import env from "react-dotenv";

 /**
    * Creates each profile displayed on the dashboard.
    * @param {user} user holds all of the info for the user.
    * @return {GetPage}
    */
function DashLeaderboardProfiles (props) {
    const [profilePic, setProfilePic] = useState("")

    axios.defaults.withCredentials = true;

    useEffect(() => { if (props.userid) axios.get(`${process.env.REACT_APP_BASE_URL}/api/getProfilePicture`, { params: { id: props.userid }} ).then((response) => { setProfilePic(response.data.profileImage) }); })

    return (
        <>
            <Card className="cardBackgroundDash" style={{ flexDirection: 'row' }}>
                <Col xs={2} lg>
                    <div className="dashLeaderNumberPosition">{props.index}.</div>
                </Col>
                <Col xs={2} lg>
                    <Image className="profileImageDash" src={`data:image/png;base64,${profilePic}`} alt="" roundedCircle />
                </Col>
                <Col xs={4} lg>
                    <div className="userTextDash">{props.name}</div>
                </Col>
                <Col xs={3} lg> 
                    <div body className="pointsDash">points: {props.score}</div> 
                 </Col>   
               
            </Card>
        </>
    );
}

export default DashLeaderboardProfiles;