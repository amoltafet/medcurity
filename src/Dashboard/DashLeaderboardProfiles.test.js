import renderer from 'react-test-renderer';
import DashLeaderboardProfiles from './DashLeaderboardProfiles';
import React from 'react';

test('renders LoginPage', () => {
  const tree = renderer.create(<DashLeaderboardProfiles />).toJSON();
  expect(tree).toMatchSnapshot();
});