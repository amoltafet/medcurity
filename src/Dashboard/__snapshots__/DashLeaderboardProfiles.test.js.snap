// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage 1`] = `
<div
  className="cardBackgroundDash card"
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <div
    className="col-lg col-2"
  >
    <div
      className="dashLeaderNumberPosition"
    >
      .
    </div>
  </div>
  <div
    className="col-lg col-2"
  >
    <img
      alt=""
      className="profileImageDash rounded-circle"
      src="data:image/png;base64,"
    />
  </div>
  <div
    className="col-lg col-4"
  >
    <div
      className="userTextDash"
    />
  </div>
  <div
    className="col-lg col-3"
  >
    <div
      body={true}
      className="pointsDash"
    >
      points: 
    </div>
  </div>
</div>
`;
